cluster-autoscaler/cloudprovider/digitalocean/digitalocean_cloud_provider.go

```diff
@@ -70,13 +71,8 @@ func (d *digitaloceanCloudProvider) NodeGroups() []cloudprovider.NodeGroup {
 // should not be processed by cluster autoscaler, or non-nil error if such
 // occurred. Must be implemented.
 func (d *digitaloceanCloudProvider) NodeGroupForNode(node *apiv1.Node) (cloudprovider.NodeGroup, error) {
-	nodeID, ok := node.Labels[nodeIDLabel]
-	if !ok {
-		// CA creates fake node objects to represent upcoming VMs that haven't
-		// registered as nodes yet. They have node.Spec.ProviderID set. Use
-		// that as nodeID.
-		nodeID = node.Spec.ProviderID
-	}
+	providerID := node.Spec.ProviderID
+	nodeID := strings.TrimPrefix(providerID, doProviderIDPrefix)
```

Comment:
Since this PR adds a `toProviderID` function already, it might be nice to also add a `toDropletID` (or `toNodeID`?) function for consistency and encapsulation reasons.